File: src/Software Engineer Prepare Kit/19/solution.go
package main

func GetAutoSaveInterval(n int32) int64 {
	if n < 2 {
		return int64(n + 1)
	}

	sequence := [2]int64{1, 2}

	for range n - 1 {
		sum := sequence[0] + sequence[1]
		sequence[0] = sequence[1]
		sequence[1] = sum
	}

	return sequence[1]
}

File: src/Software Engineer Prepare Kit/19/solution.ts
export default function getAutoSaveInterval(n: number): number {
	if (n === 0) return BigInt(1) as unknown as number
	if (n === 1) return BigInt(2) as unknown as number

	const sequence: [bigint, bigint] = [BigInt(1), BigInt(2)]

	for (let i = 2; i <= n; i++) {
		const sum = sequence[0] + sequence[1]
		sequence[0] = sequence[1]
		sequence[1] = sum
	}

	return sequence[1] as unknown as number
}
